### ISSUE TITLE
Provide access to raw ErrorResponse body

### ISSUE THREAD

**Issue by jayschwa** [xanzy/go-gitlab#400]
`ErrorResponse.Message` isn't formatted in a way that is useful to me. It's not very human-readable, and it's not encoded in a standard format either. I would like to have access to the raw JSON. Unfortunately, it can no longer be obtained from the `ErrorResponse.Response.Body` because it has already been read. I propose saving the raw JSON into a new field on the `ErrorResponse` so that it is accessible.

**Comment by zaquestion** [comment]
@jayschwa just to clarify, `[]interface{}` and `map[string]interface{}` aren't useful to you? When you print maps or strings in go (without something like spew) they can looks pretty hard to read for humans, but the structures should be easy to work with. These are the types `ErrorResponse.Message` can be https://github.com/xanzy/go-gitlab/blob/master/gitlab.go#L707

**Comment by zaquestion** [comment]
@jayschwa my bad, probably shouldn't have been reading code in the middle of the night. That function does just return a string as you said, I thought it returned the types it was switching on for some silly reason.

In that case, thoughts on putting the raw response on an ErrorResponse.Raw field. You'd have to type assert it out to use it. For example:
```
_, err := Do(...)
if er, ok := err.(gitlab.ErrorResponse); ok {
    _ = er.Raw
}
```

Not sure how I feel about the field name, but putting the unparsed type on that field seems reasonable to me.

**Comment by jayschwa** [comment]
I think `ErrorResponse.Body` accurately describes the hypothetical field.

**Comment by svanharmelen** [comment]
Should be fixed by PR #422
